src/frontends/onnx/frontend/src/input_model.cpp

```diff
@@ -114,8 +114,22 @@ void InputModel::set_name_for_dimension(const ov::frontend::Place::Ptr& tensor,
     onnx_tensor->set_name_for_dimension(shape_dim_index, dim_name);
 }
 
-void InputModel::add_name_for_tensor(const ov::frontend::Place::Ptr&, const std::string&) {
-    FRONT_END_THROW("Method add_name_for_tensor is not applicable for ONNX model. ONNX tensor has just one name.");
+void InputModel::add_name_for_tensor(const ov::frontend::Place::Ptr& tensor, const std::string& new_name) {
+    ov::frontend::Place::Ptr tensor_place = tensor;
+    const auto input_edge = std::dynamic_pointer_cast<PlaceInputEdge>(tensor);
+    if (input_edge) {
+        tensor_place = input_edge->get_source_tensor();
+    }
+
+    const auto onnx_tensor = std::dynamic_pointer_cast<PlaceTensor>(tensor_place);
+    FRONT_END_GENERAL_CHECK(onnx_tensor != nullptr,
+                            "Incorrect Place passed to add_name_for_tensor. This method expects a PlaceTensor object "
+                            "pointing to the ONNX tensor.");
+    FRONT_END_GENERAL_CHECK(!onnx_tensor->get_names().empty(),
```

Comment:
Is this really required to have some name? Can there be a case that tensor "some_node:2" doesn't have any tensor name yet, but here we should be able to assign it

Reply:
The reason I added this check is because I used unsafe accessing of the returned vector using `operator[]`. I'll remove this check because this vector cannot be empty.